balancer/pickfirst/pickfirst_ext_test.go

```diff
@@ -2082,6 +2117,86 @@ func (s) TestPickFirstLeaf_HappyEyeballs_TF_ThenTimerFires(t *testing.T) {
 	}
 }
 
+// Test verifies that when a subchannel is shut down by the LB (because another
+// subchannel won) while its dial is still in-flight, it records exactly one
+// successful attempt.
+func (s) TestPickFirstLeaf_HappyEyeballs_Ignore_Inflight_Cancellations(t *testing.T) {
+	t.Log("starting test")
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	originalTimer := pfinternal.TimeAfterFunc
+	defer func() {
+		pfinternal.TimeAfterFunc = originalTimer
+	}()
+	triggerTimer, timeAfter := mockTimer()
+	pfinternal.TimeAfterFunc = timeAfter
+
+	tmr := stats.NewTestMetricsRecorder()
+	dialer := testutils.NewBlockingDialer()
+	opts := []grpc.DialOption{
+		grpc.WithDefaultServiceConfig(fmt.Sprintf(`{"loadBalancingConfig": [{"%s":{}}]}`, pfbalancer.Name)),
+		grpc.WithContextDialer(dialer.DialContext),
+		grpc.WithStatsHandler(tmr),
+	}
+
+	// Setup 2 backend addresses
+	cc, rb, bm := setupPickFirstLeaf(t, 2, opts...)
+	addrs := bm.resolverAddrs()
+	holds := bm.holds(dialer)
+	rb.UpdateState(resolver.State{Addresses: addrs})
+	cc.Connect()
+
+	// Make sure we conncet to second subconn
+	testutils.AwaitState(ctx, t, cc, connectivity.Connecting)
+	if holds[0].Wait(ctx) != true {
+		t.Fatalf("Timeout waiting for server %d to be contacted", 0)
+	}
+	triggerTimer()
+	if holds[1].Wait(ctx) != true {
+		t.Fatalf("Timeout waiting for server %d to be contacted", 1)
+	}
+	holds[1].Resume()
+
+	// Wait for Channel to become READY.
+	testutils.AwaitState(ctx, t, cc, connectivity.Ready)
+
+	// Unblock the First SubConn.
+	// Since the LB has already closed this subchannel, the context passed to Dial
+	// is canceled. This will lead to an inflight attempt to be cancelled.
+	// No success or failure metric should be recorded for this.
+	holds[0].Resume()
+
+	// --- Assertions ---
+
+	// Wait for the SUCCESS metric to ensure recording logic has processed.
+	waitForMetric(ctx, t, tmr, "grpc.subchannel.connection_attempts_succeeded")
+
+	// Verify Success: Exactly 1 (The Winner).
+	if got, _ := tmr.Metric("grpc.subchannel.connection_attempts_succeeded"); got != 1 {
+		t.Errorf("Unexpected data for metric %v, got: %v, want: 1", "grpc.subchannel.connection_attempts_succeeded", got)
+	}
+
+	// Verify Failure: Exactly 0 (The Loser was ignored).
+	// We poll briefly to ensure no delayed failure metric appears.
+	shortCtx, shortCancel := context.WithTimeout(ctx, 50*time.Millisecond)
```

Comment:
Nit: The commonly used variable names for this are `sCtx` and `sCancel`. Go generally prefers shorter variable names where appropriate. See: go/go-style/decisions#variable-names